druid/src/contexts.rs

```diff
@@ -315,6 +316,17 @@ impl_context_method!(EventCtx<'_, '_>, UpdateCtx<'_, '_>, LifeCycleCtx<'_, '_>,
     pub fn set_menu<T: Any>(&mut self, menu: MenuDesc<T>) {
         self.state.set_menu(menu);
     }
+
+    /// Create a new sub window that will have its app data synchronised with the nearest surrounding widget pod.
+    // TODO - dynamically check that the type of the pod we are registering this on is the same as the type of the
+    // requirement. Needs type ids recorded. This goes wrong if you don't have a pod between you and a lens.
+    pub fn new_sub_window(&mut self, requirement: SubWindowRequirement) {
```

Comment:
Thinking about this, it feels like we could maybe simplify?

Basically the only things I think we *need* to provide here is a `WindowConfig` and some builder for the root widget.

we don't need the `host_id`, because we can just get that here ourselves from the context. I could imagine also taking a flag that indicates whether or not we take updates. I also think asking the user to pass in the `WindowId` might be unnecessary; could we just create a new `WindowId` ourselves, pass it along with the `NEW_SUB_WINDOW` command, and then also return a copy to the caller?

So what I would maybe imagine:

rename `SubWindowRequirement` to just `SubWindow`; this will be the user-facing, public API. Then we can either have a separate `SubWindowArgs` struct that is 100% internal and is the payload of the command, *or* we can just have some private fields on `SubWindow` that we set internally.

To be clear, this isn't a change in any behaviour; my motivation here is 100% ergonomics.

Reply:
I managed to make it just the method on context that is the api. The rest is private now.